src/Ryujinx.Common/GraphicsDriver/DriverUtilities.cs

```
@@ -1,21 +1,26 @@
+using Ryujinx.Common.GraphicsDriver.NVAPI;
+using Ryujinx.Common.Logging;
 using System;
 
 namespace Ryujinx.Common.GraphicsDriver
 {
     public static class DriverUtilities
     {
-        public static void ToggleOGLThreading(bool enabled)
+        public static void ToggleNvDriverSetting(NvapiSettingId id, bool enabled)
         {
-            Environment.SetEnvironmentVariable("mesa_glthread", enabled.ToString().ToLower());
-            Environment.SetEnvironmentVariable("__GL_THREADED_OPTIMIZATIONS", enabled ? "1" : "0");
-
             try
             {
-                NVThreadedOptimization.SetThreadedOptimization(enabled);
+                if (id == NvapiSettingId.OglThreadControlId)
+                {
+                    Environment.SetEnvironmentVariable("mesa_glthread", enabled.ToString().ToLower());
```

These settings are not for nvidia, yet the function has been renamed with "nv" in the name. I'm not sure if these are actually doing anything right now, due to how env vars can be cached. (maybe if the change happens before the driver is loaded...?)

---

I suppose this is still somewhat tied to how this whole chain used to be setup.
Threading is currently toggled from: https://github.com/Ryujinx/Ryujinx/blob/c6f8bfed904e30f7c5d890a2f0ef531eb9e298e5/src/Ryujinx/Program.cs#L115 which sets these environment variables and _then_ goes into the NVAPI section.

In order to generalize how NVCP settings I instead chose to pass the NV threading ID to denote "Threaded optimizations" more generally in this one case (hence the if).

---

Yeah, but the name for that method would probably be `ToggleOGLThreading`, which more indicates that the nv setting change should be called from this method rather than replacing it.

---

Right I see what you mean. Would this work better

```cs
public static class DriverUtilities
    {
        public static void ToggleOglThreading(bool enabled)
        {
            Environment.SetEnvironmentVariable("mesa_glthread", enabled.ToString().ToLower());
            Environment.SetEnvironmentVariable("__GL_THREADED_OPTIMIZATIONS", enabled ? "1" : "0");

            ToggleNvDriverSetting(NvapiSettingId.OglThreadControlId, enabled);
        }
        
        public static void ToggleNvDriverSetting(NvapiSettingId id, bool enabled)
        {
            try
            {
                NVDriverHelper.SetControlOption(id, enabled);
            }
            catch
            {
                Logger.Warning?.Print(LogClass.Application, "Failed to set NVIDIA driver settings. NVAPI may be unavailable.");
            }
        }
    }
```

---

Yeah, I think that would cover this case well. I'm not sure we should be printing a warning here, since it will always fail on AMD.

---

Alright I've swapped this up. Moved to a less aggressive info log because I really don't like an empty catch here. 
I see little harm with this being info'd when we have stuff like Hypervisor and Colour space passthrough which do the same on unsupported systems.

Quick change if you insist though.